Cannot build audiowaveform on Windows WSL2 for arm64

Hi, I have problems building the `audiowaveform` part on Windows 11 with WSL2 (for the linux/arm64 amazonlinux:2 platform - the amd64 version seems to work).

"configure: error: cannot guess build type; you must specify one"
Should this work on WSL2?

```shell
user@<redacted>:/mnt/c/repositories/audiotool-builders/audiowaveform$ docker build -t audiowaveform .
[+] Building 6.8s (10/10) FINISHED                                                                                                                                                                      docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                              0.0s
 => => transferring dockerfile: 386B                                                                                                                                                                              0.0s 
 => [internal] load metadata for docker.io/library/amazonlinux:2                                                                                                                                                  0.8s 
 => [internal] load .dockerignore                                                                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                                                                   0.0s 
 => [1/6] FROM docker.io/library/amazonlinux:2@sha256:b8607887eebccfafa27494fc22841db4f1d707e506b0df293077dc1af2d027d0                                                                                            0.0s 
 => [internal] load build context                                                                                                                                                                                 0.0s 
 => => transferring context: 30B                                                                                                                                                                                  0.0s 
 => CACHED [2/6] RUN yum -y update                                                                                                                                                                                0.0s 
 => CACHED [3/6] RUN yum -y install make cmake3 autogen automake libtool gcc gcc-c++ wget tar   gzip zip libcurl-devel zlib-static libpng-static xz git python-devel   bzip2-devel which gd-devel                 0.0s 
 => CACHED [4/6] COPY ./build.sh build.sh                                                                                                                                                                         0.0s
 => CACHED [5/6] RUN chmod +x build.sh                                                                                                                                                                            0.0s 
 => ERROR [6/6] RUN ./build.sh                                                                                                                                                                                    5.8s 
------
 > [6/6] RUN ./build.sh:
0.360 + wget https://netix.dl.sourceforge.net/project/mad/libid3tag/0.15.1b/libid3tag-0.15.1b.tar.gz
0.397 --2024-02-28 15:57:27--  https://netix.dl.sourceforge.net/project/mad/libid3tag/0.15.1b/libid3tag-0.15.1b.tar.gz
0.447 Resolving netix.dl.sourceforge.net (netix.dl.sourceforge.net)... 87.121.121.2
0.497 Connecting to netix.dl.sourceforge.net (netix.dl.sourceforge.net)|87.121.121.2|:443... connected.
0.665 HTTP request sent, awaiting response... 200 OK
0.738 Length: 338143 (330K) [application/x-gzip]
0.739 Saving to: 'libid3tag-0.15.1b.tar.gz'
0.741
0.741      0K .......... .......... .......... .......... .......... 15%  511K 1s
0.838     50K .......... .......... .......... .......... .......... 30%  784K 0s
0.901    100K .......... .......... .......... .......... .......... 45%  399K 0s
1.027    150K .......... .......... .......... .......... .......... 60%  791K 0s
1.090    200K .......... .......... .......... .......... .......... 75%  744K 0s
1.157    250K .......... .......... .......... .......... .......... 90%  784K 0s
1.221    300K .......... .......... ..........                      100% 1.19M=0.5s
1.247
1.250 2024-02-28 15:57:28 (652 KB/s) - 'libid3tag-0.15.1b.tar.gz' saved [338143/338143]
1.250
1.255 + tar xzf libid3tag-0.15.1b.tar.gz
1.315 + cd libid3tag-0.15.1b
1.317 + sed -i 's/ -fforce-mem//' configure
1.399 + ./configure --disable-shared --libdir=/lib64
2.551 checking for a BSD-compatible install... /usr/bin/install -c
2.597 checking whether build environment is sane... yes
3.778 checking for gawk... gawk
3.778 checking whether make sets $(MAKE)... yes
4.150 checking build system type... ./config.guess: unable to guess system type
5.268
5.268 This script, last modified 2004-01-05, has failed to recognize
5.268 the operating system you are using. It is advised that you
5.268 download the most up to date version of the config scripts from
5.268
5.268     ftp://ftp.gnu.org/pub/gnu/config/
5.268
5.268 If the version you run (./config.guess) is already up to date, please
5.268 send the following data and any information you think might be
5.268 pertinent to <config-patches@gnu.org> in order to provide the needed
5.268 information to handle your system.
5.268
5.268 config.guess timestamp = 2004-01-05
5.268
5.268 uname -m = aarch64
5.268 uname -r = 5.15.133.1-microsoft-standard-WSL2
5.268 uname -s = Linux
5.268 uname -v = #1 SMP Thu Oct 5 21:02:42 UTC 2023
5.268
5.268 /usr/bin/uname -p = aarch64
5.268 /bin/uname -X     =
5.268
5.268 hostinfo               =
5.268 /bin/universe          =
5.268 /usr/bin/arch -k       =
5.268 /bin/arch              = aarch64
5.268 /usr/bin/oslevel       =
5.268 /usr/convex/getsysinfo =
5.268
5.268 UNAME_MACHINE = aarch64
5.268 UNAME_RELEASE = 5.15.133.1-microsoft-standard-WSL2
5.268 UNAME_SYSTEM  = Linux
5.268 UNAME_VERSION = #1 SMP Thu Oct 5 21:02:42 UTC 2023
5.307 configure: error: cannot guess build type; you must specify one
------
Dockerfile:10
--------------------
   8 |     COPY ./build.sh build.sh
   9 |     RUN chmod +x build.sh
  10 | >>> RUN ./build.sh
  11 |
--------------------
ERROR: failed to solve: process "/bin/sh -c ./build.sh" did not complete successfully: exit code: 1
```

--- a/audiowaveform/Dockerfile
+++ b/audiowaveform/Dockerfile
@@ -1,5 +1,5 @@
 FROM --platform=linux/amd64 amazonlinux:2
-# FROM --platform=linux/arm64 amazonlinux:2
+# FROM --platform=linux/arm64/v8 amazonlinux:2
 RUN yum -y update
 RUN yum -y install make cmake3 autogen automake libtool gcc gcc-c++ wget tar \
   gzip zip libcurl-devel zlib-static libpng-static xz git python-devel \

--- a/audiowaveform/build.sh
+++ b/audiowaveform/build.sh
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+# When building for arm64, uncomment everything in the "./configure" command lines
+
 set -ex
 
 # Build libid3tag
@@ -7,7 +9,7 @@ wget https://netix.dl.sourceforge.net/project/mad/libid3tag/0.15.1b/libid3tag-0.
 tar xzf libid3tag-0.15.1b.tar.gz
 cd libid3tag-0.15.1b
 sed -i 's/ -fforce-mem//' configure
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -16,7 +18,7 @@ wget https://netix.dl.sourceforge.net/project/mad/libmad/0.15.1b/libmad-0.15.1b.
 tar xzf libmad-0.15.1b.tar.gz
 cd libmad-0.15.1b
 sed -i 's/ -fforce-mem//' configure
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -27,7 +29,7 @@ tar xzf 1.0.28.tar.gz
 cd libsndfile-1.0.28
 sed -i 's/flac >= 1.3.1/flac >= 1.3.0/' configure.ac
 ./autogen.sh
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -35,7 +37,7 @@ cd /
 wget https://ftp.osuosl.org/pub/xiph/releases/flac/flac-1.3.0.tar.xz
 tar xf flac-1.3.0.tar.xz
 cd flac-1.3.0
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -43,7 +45,7 @@ cd /
 wget http://downloads.xiph.org/releases/ogg/libogg-1.3.4.tar.gz
 tar xf libogg-1.3.4.tar.gz
 cd libogg-1.3.4
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -51,7 +53,7 @@ cd /
 wget http://downloads.xiph.org/releases/vorbis/libvorbis-1.3.6.tar.gz
 tar xf libvorbis-1.3.6.tar.gz
 cd libvorbis-1.3.6
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 
@@ -59,7 +61,7 @@ cd /
 wget https://archive.mozilla.org/pub/opus/opus-1.3.1.tar.gz
 tar xzf opus-1.3.1.tar.gz
 cd opus-1.3.1
-./configure --disable-shared --libdir=/lib64
+./configure --disable-shared --libdir=/lib64 # --build=aarch64-unknown-linux-gnu
 make install
 cd /
 

